### PR TITLE
erlang: add updateScript and update versions

### PR DESCRIPTION
**Pull request by dlesl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dlesl** [comment]
My hope is that this change will mean that [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) will be able to generate automatic PRs

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118831` at d5ca42d6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>couchdb</li>
    <li>couchdb2</li>
    <li>cuter</li>
    <li>erlangR18</li>
    <li>erlangR19</li>
    <li>lfe_1_2</li>
    <li>wings</li>
    <li>yaws</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/749fb692f152d3e5e4e54ed3f07244ea">couchdb3</a></li>
  </ul>
</details>
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>erlangR20</li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>asls</li>
    <li>erlangR21</li>
    <li>erlangR22</li>
    <li>lfe (lfe_1_3)</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

File: pkgs/development/interpreters/erlang/R23.nix
Comment (SuperSandro2000) [inline pkgs/development/interpreters/erlang/R23.nix:7]: Why did this hash change?

File: pkgs/development/interpreters/erlang/generic-builder.nix
Comment (SuperSandro2000) [inline pkgs/development/interpreters/erlang/generic-builder.nix:120]:
```suggestion
        lib.makeBinPath [ common-updater-scripts coreutils git gnused ]
```

File: pkgs/development/interpreters/erlang/R23.nix
Comment (dlesl) [reply pkgs/development/interpreters/erlang/R23.nix:7]: As a result of running the script, I guess it's the same hash but in the new format. I'll change it back!

Comment (dlesl) [reply pkgs/development/interpreters/erlang/R23.nix]: The script shouldn't change the hash if the version hasn't increased, actually, what happen is that I replaced this file with an older version to check that it was able to update it, and it ended up with the new hash format.

File: pkgs/development/interpreters/erlang/generic-builder.nix
Comment (yurrriq) [reply pkgs/development/interpreters/erlang/generic-builder.nix]: I'm fairly strongly in favor of multiple lines here as it makes future diffs easier to reason about. Evan, the author of Elm, had a good blog post about this a while back, but I can't seem to find it quickly. Can you help me understand your reasoning behind this suggestion?

File: pkgs/development/interpreters/erlang/generic-builder.nix
Comment (yurrriq) [inline pkgs/development/interpreters/erlang/generic-builder.nix:126]: Thanks for this! I did not test it.

**Review by yurrriq — APPROVED** [review]

File: pkgs/development/interpreters/erlang/generic-builder.nix
Comment (SuperSandro2000) [reply pkgs/development/interpreters/erlang/generic-builder.nix]: I wanted to shorten the amount of lines to make the derivation not bigger as it is. If you like it this way out is fine for me, too and it won't block the merge.

**Comment by happysalada** [comment]
Regarding formatting issues, is there a way we can agree on a formatter and stick to whatever it says?
I would suggest https://github.com/nix-community/nixpkgs-fmt

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 118831` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>couchdb</li>
    <li>couchdb2</li>
    <li>couchdb3</li>
    <li>cuter</li>
    <li>erlangR18</li>
    <li>erlangR19</li>
    <li>lfe_1_2</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>asls</li>
    <li>erlangR20</li>
    <li>erlangR21</li>
    <li>erlangR22</li>
    <li>lfe (lfe_1_3)</li>
  </ul>
</details>

**Review by happysalada — APPROVED** [review]
I would prefer using nixpkgs-fmt, but no matter what you decide, this should not block the merge.

**Comment by happysalada** [comment]
@balsoft I think this is good to merge, can you take a look perhaps when you have a moment?

File: pkgs/development/interpreters/erlang/generic-builder.nix
Comment (SuperSandro2000) [inline pkgs/development/interpreters/erlang/generic-builder.nix:116]:
```suggestion
      PATH=${lib.makeBinPath [ common-updater-scripts coreutils git gnused ]}
```

**Pull request merged**
